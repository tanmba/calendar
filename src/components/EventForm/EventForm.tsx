import React, { ChangeEventHandler, MouseEventHandler, SyntheticEvent } from "react";
import { Box, TextField } from "@mui/material";
import DatePicker from "react-datepicker";
import {setMinutes, setHours} from "date-fns";

interface EventFormProps {
  /**
   * Event title.
   */
  title?: string;
  /**
   * Event vendor info.
   */
  vendor?: string;
  /**
   * Event buyer info.
   */
  buyer?: string;
  /**
   * Event start date.
   */
  start?: Date;
  /**
   * Event end date.
   */
  end?: Date;
  /**
   * On form title change function.
   */
  onTitleChange?: ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement>;
  /**
   * On form vendor change function.
   */
  onVendorChange?: ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement>;
  /**
   * On form buyer change function.
   */
  onBuyerChange?: ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement>;
  /**
   * On form start date change function.
   */
  onStartChange?: any;
  /**
   * On form end date change function.
   */
  onEndChange?: any;
  /**
   * Add event function.
   */
  addEvent?: MouseEventHandler<HTMLButtonElement>;
}

const EventForm = ({
  title, 
  vendor, 
  buyer, 
  start, 
  end, 
  onTitleChange, 
  onBuyerChange, 
  onVendorChange, 
  onStartChange, 
  onEndChange, 
  addEvent}: EventFormProps) => {

  const now = new Date();

  //Text field container style
  const textFieldContainerStyle = {
    width: '100%', 
    display: 'flex', 
    justifyContent: 'space-between',
    marginBottom: '2rem'
  }

  return (
    <React.Fragment>
      <Box component="h2" sx={{ textAlign: "center" }}>Add event</Box>
      <Box sx={{ display: 'flex', flexDirection: 'column' }}>
        <TextField
          sx={{ width: '100%', marginBottom: '2rem'}}
          id="outlined-basic" 
          label="Event name" 
          variant="outlined"
          value={title} 
          onChange={onTitleChange}
        />
        <Box sx={textFieldContainerStyle}>
          <TextField
            sx={{ width: '47%'}}
            id="outlined-basic" 
            label="Vendor" 
            variant="outlined"
            value={vendor}
            onChange={onVendorChange}
          />
          <TextField
            sx={{ width: '47%'}}
            id="outlined-basic" 
            label="Buyer" 
            variant="outlined"
            value={buyer}
            onChange={onBuyerChange}
          />
        </Box>
        <Box sx={textFieldContainerStyle}>
          <Box>
            <Box sx={{ marginBottom: "0.5rem" }}>From</Box>
            <DatePicker
              dateFormat="MM/d/yyyy h:mm aa"
              customInput={<TextField />}
              timeIntervals={15} 
              minDate={now}
              minTime={setHours(setMinutes(new Date(), 0), 9)}
              maxTime={setHours(setMinutes(new Date(), 0), 18)}
              showTimeSelect
              placeholderText="Start Date" 
              selected={start} 
              onChange={onStartChange} 
            />
          </Box>
          <Box>
            <Box sx={{ marginBottom: "0.5rem" }}>To</Box>
            <DatePicker
              dateFormat="MM/d/yyyy h:mm aa"
              customInput={<TextField />}
              timeIntervals={15} 
              minDate={now}
              minTime={setHours(setMinutes(new Date(), 0), 9)}
              maxTime={setHours(setMinutes(new Date(), 0), 18)} 
              showTimeSelect 
              placeholderText="End Date" 
              selected={end} 
              onChange={onEndChange} 
            />
          </Box>
        </Box>
      </Box>
      <Box component="button" sx={{ marginTop: "10px" }} onClick={addEvent}>
        Add Event
      </Box>
    </React.Fragment>
  )
}

export default EventForm;