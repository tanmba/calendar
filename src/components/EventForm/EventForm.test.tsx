import React from "react";
import EventForm from "./EventForm";
import { render } from "@testing-library/react";

describe("Event form component", () => {
  it("renders Event form component correctly", () => {
    const { asFragment } = render(<EventForm />);
    expect(asFragment()).toMatchSnapshot();
  });
});