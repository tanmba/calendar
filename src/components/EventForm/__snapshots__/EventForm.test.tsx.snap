// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event form component renders Event form component correctly 1`] = `
<DocumentFragment>
  <h2
    class="MuiBox-root css-xi606m"
  >
    Add event
  </h2>
  <div
    class="MuiBox-root css-j7qwjs"
  >
    <div
      class="MuiFormControl-root MuiTextField-root css-a1xxqq-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for="outlined-basic"
        id="outlined-basic-label"
      >
        Event name
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id="outlined-basic"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-yjsfm1"
          >
            <span>
              Event name
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      class="MuiBox-root css-1ssq1cy"
    >
      <div
        class="MuiFormControl-root MuiTextField-root css-1ci2vmj-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="outlined-basic"
          id="outlined-basic-label"
        >
          Vendor
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id="outlined-basic"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                Vendor
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root css-1ci2vmj-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="outlined-basic"
          id="outlined-basic-label"
        >
          Buyer
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id="outlined-basic"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                Buyer
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root css-1ssq1cy"
    >
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="MuiBox-root css-cet0rr"
        >
          From
        </div>
        <div
          class="react-datepicker-wrapper"
        >
          <div
            class="react-datepicker__input-container"
          >
            <div
              class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":r3:"
                  placeholder="Start Date"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-ihdtdm"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="MuiBox-root css-cet0rr"
        >
          To
        </div>
        <div
          class="react-datepicker-wrapper"
        >
          <div
            class="react-datepicker__input-container"
          >
            <div
              class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":r4:"
                  placeholder="End Date"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-ihdtdm"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <button
    class="MuiBox-root css-1q7njkh"
  >
    Add Event
  </button>
</DocumentFragment>
`;
